tests/infrastructure/test_data_manager_stack.py

```diff
@@ -26,3 +32,824 @@ def test_sds_data_manager_stack(app, sds_id):
+
+
+def test_query_api_lambda_url_resource_properties(template):
+    template.has_resource_properties(
+        "AWS::Lambda::Url",
+        {
+            "AuthType": "NONE",
+            "TargetFunctionArn": {
+                "Fn::GetAtt": [Match.string_like_regexp("QueryAPILambda*"), "Arn"]
+            },
+            "Cors": {"AllowMethods": ["GET"], "AllowOrigins": ["*"]},
+        },
+    )
+
+
+def test_download_api_lambda_url_resource_properties(template):
+    template.has_resource_properties(
+        "AWS::Lambda::Url",
+        {
+            "AuthType": "NONE",
+            "TargetFunctionArn": {
+                "Fn::GetAtt": [
+                    Match.string_like_regexp("DownloadQueryAPILambda*"),
+                    "Arn",
+                ]
+            },
+            "Cors": {"AllowMethods": ["GET"], "AllowOrigins": ["*"]},
+        },
+    )
+
+
+def test_lambda_function_resource_count(template):
+    template.resource_count_is("AWS::Lambda::Function", 7)
+
+
+def test_indexer_lambda_function_resource_properties(template, sds_id):
+    template.has_resource_properties(
+        "AWS::Lambda::Function",
+        props={
+            "FunctionName": f"file-indexer-{sds_id}",
+            "Runtime": "python3.9",
+            "Handler": "SDSCode.indexer.lambda_handler",
+            "MemorySize": 1000,
+            "Timeout": 15 * 60,
+        },
+    )
+
+
+def test_upload_api_lambda_function_resource_properties(template, sds_id):
+    template.has_resource_properties(
+        "AWS::Lambda::Function",
+        props={
+            "FunctionName": f"upload-api-handler-{sds_id}",
+            "Runtime": "python3.9",
+            "Handler": "SDSCode.upload_api.lambda_handler",
+            "MemorySize": 1000,
+            "Timeout": 15 * 60,
+        },
+    )
+
+
+def test_query_api_lambda_function_resource_properties(template, sds_id):
+    template.has_resource_properties(
+        "AWS::Lambda::Function",
+        props={
+            "FunctionName": f"query-api-handler-{sds_id}",
+            "Runtime": "python3.9",
+            "Handler": "SDSCode.queries.lambda_handler",
+            "MemorySize": 1000,
+            "Timeout": 60,
+        },
+    )
+
+
+def test_download_api_lambda_function_resource_properties(template, sds_id):
+    template.has_resource_properties(
+        "AWS::Lambda::Function",
+        props={
+            "FunctionName": f"download-query-api-{sds_id}",
+            "Runtime": "python3.9",
+            "Handler": "SDSCode.download_query_api.lambda_handler",
+            "Timeout": 60,
+        },
+    )
+
+
+def test_aws_lambda_function_resource_properties(template):
+    template.has_resource_properties(
+        "AWS::Lambda::Function",
+        props={
+            "Handler": "index.handler",
+            "Runtime": "nodejs14.x",
+            "Timeout": 120,
+            "Role": {
+                "Fn::GetAtt": [
+                    "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
```

Comment:
Do we want to assert this specific hash role, or just leave it as a `Match.Any()` / remove the role check altogether?

Reply:
I'll remove the Role check